commerce_coordinator/apps/lms/filters.py

```diff
@@ -33,17 +33,15 @@ class OrderRefundRequested(OpenEdxPublicFilter):
     filter_type = "org.edx.coordinator.lms.order.refund.requested.v1"
 
     @classmethod
-    def run_filter(cls, order_number, order_line_id):  # pragma no cover
-        # TODO: Filter will be called in SONIC-83
+    def run_filter(cls, order_id, order_line_id):  # pragma no cover
         """
         Call the PipelineStep(s) defined for this filter.
         Arguments:
-            order_number: Order number (for now this is an order.id, but this should change in the future)
-            TODO: SONIC-277 (in-progress)
+            order_id: Order Id
             order_line_id: ID of order line item
         Returns:
             order_refund: Updated order with return item attached
         """
 
-        order_refund = super().run_pipeline(order_number=order_number, order_line_id=order_line_id)
+        order_refund = super().run_pipeline(order_id=order_id, order_line_id=order_line_id)
```

Comment:
```suggestion
        order_refund = super().run_pipeline(order_number=order_id, order_line_id=order_line_id)
```
Argument naming inconsistency between pipeline steps:

If we look at the pipeline steps [here](https://github.com/edx/commerce-coordinator/blob/ca054ba93b3d612ea740b5b4868ae96adafbc9c7/commerce_coordinator/settings/base.py#L377-L385), we are calling 2 pipelines and first step's argument name is [order_number](https://github.com/edx/commerce-coordinator/blob/ca054ba93b3d612ea740b5b4868ae96adafbc9c7/commerce_coordinator/apps/rollout/pipeline.py#L76).

And, in our second pipeline step, we have renamed the argument to `order_id` in this PR. I am not sure if this pipeline filter can pass the argument correctly and work properly. You might need to test this.

Reply:
I've renamed the second pipeline step back to order_number, for consistency and because I misinterpreted the pipeline argument code comment. It is now back to order_number

Reply:
It seems like you may not have a clear understanding of the code flow. Let me explain how this pipeline works:

- [OrderRefundRequested](https://github.com/edx/commerce-coordinator/blob/ca054ba93b3d612ea740b5b4868ae96adafbc9c7/commerce_coordinator/apps/lms/views.py#L225) gets called in our `RefundView` endpoint. It takes two arguments, `order_id` and `order_line_id`. These are arguments that were originally passed from LMS's enrollment attributes table. `order_id` argument has a value of CT's order's ID (UUID), not the human-readable order number

- If we look at the pipeline steps [here](https://github.com/edx/commerce-coordinator/blob/ca054ba93b3d612ea740b5b4868ae96adafbc9c7/commerce_coordinator/settings/base.py#L377-L385), we are calling 2 pipelines

- first pipeline [step](https://github.com/edx/commerce-coordinator/blob/ca054ba93b3d612ea740b5b4868ae96adafbc9c7/commerce_coordinator/apps/rollout/pipeline.py#L76) has its first argument named as `order_number` (I am not sure the reason, but it is used in multiple pipelines) but it stores the CT Order's ID (UUID) as a value. This is where all the confusion starts from. We should either change its argument name to `order_id` and test all the filters where this pipeline is used to ensure that it is working fine or create a new pipeline step for this view for the same purpose and pass `order_id` as the argument name

- For the above reason, I suggested to keep the [run_pipline()](https://github.com/edx/commerce-coordinator/pull/207/files?diff=unified&w=1#r1580658992) function's argument name consistent `order_number`

- second pipeline [step](https://github.com/edx/commerce-coordinator/blob/ca054ba93b3d612ea740b5b4868ae96adafbc9c7/commerce_coordinator/apps/commercetools/pipeline.py#L114) has arguments passed from previous (first pipeline) and the argument `order_number` it has stores the CT Order's ID (UUID) as a value. This is the value passed to it from the view from where this pipeline filter is called from

I hope you have got the code flow understanding now. LMK, if you need to sync on this.